charts/argocd-progressive-rollout/templates/deployment.yaml

```diff
@@ -0,0 +1,57 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "argocd-progressive-rollout.fullname" . }}
+  labels:
+    {{- include "argocd-progressive-rollout.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "argocd-progressive-rollout.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "argocd-progressive-rollout.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "argocd-progressive-rollout.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          command:
+            - /manager
+            - --metrics-addr={{ .Values.args.metricsAddr }}
+            {{- if or (gt ( .Values.replicaCount | int64) 1) .Values.args.enableLeaderElection }}
+            - --enable-leader-election=true
+            {{- end }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http-metrics
+              containerPort: {{ (split ":" .Values.args.metricsAddr)._1 }}
```

Comment:
Would it be better to have two params for hostname and port in the values file instead of having to split here?

Reply:
I was doing this because then we have 1:1 match between the chart arguments and the controller flags. I'd prefer keep it this way unless you have a strong objection.

Reply:
No, not a strong objection - I saw that in other places, ie for image we have separate fields so I was wondering why this would be different.